.github/workflows/publish_test_results.yml

```diff
@@ -0,0 +1,61 @@
+name: Publish Test Results
+
+on:
+   workflow_run:
+      workflows: ["CI"]
+      types:
+         - completed
+
+jobs:
+  test-results:
+    runs-on: ubuntu-latest
+    name: Publish Test Results
+    if: >
+        github.event.workflow_run.conclusion != 'skipped' && (
+          github.event.workflow_run.head_repository.full_name != github.repository
+        )
+    steps:
+      - name: Download Artifacts
+        uses: actions/github-script@v3.1.0
+        with:
+          script: |
+              var fs = require('fs');
+              var path = require('path');
+              var artifacts_path = path.join('${{github.workspace}}', 'artifacts')
+              fs.mkdirSync(artifacts_path, { recursive: true })
+
+              var artifacts = await github.actions.listWorkflowRunArtifacts({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                run_id: ${{ github.event.workflow_run.id }},
+              });
+
+              for (const artifact of artifacts.data.artifacts) {
+                var download = await github.actions.downloadArtifact({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    artifact_id: artifact.id,
+                    archive_format: 'zip',
+                });
+                var artifact_path = path.join(artifacts_path, `${artifact.name}.zip`)
+                fs.writeFileSync(artifact_path, Buffer.from(download.data));
+                console.log(`Downloaded ${artifact_path}`);
+              }
```

Comment:
Is it possible to have that as a standalone file instead of inlining in YAML?

Reply:
No, we cannot checkout code in this workflow.

Reply:
Also to be fair, I have no idea if this will work and no good way to test it here because the workflow won't run until its merged :(

Reply:
I see. It might be better to package that as our own GitHub actions, in future.